datasets/wi_locness/wi_locness.py

```diff
@@ -0,0 +1,222 @@
+submissions and assigned them a CEFR level.
+
+The LOCNESS corpus (Granger, 1998) consists of essays written by native English students.
+It was originally compiled by researchers at the Centre for English Corpus Linguistics at the
+University of Louvain. Since native English students also sometimes make mistakes, we asked
+the W&I annotators to annotate a subsection of LOCNESS so researchers can test the effectiveness
+of their systems on the full range of English levels and abilities."""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+from pathlib import Path
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{bryant-etal-2019-bea,
+    title = "The {BEA}-2019 Shared Task on Grammatical Error Correction",
+    author = "Bryant, Christopher  and
+        Felice, Mariano  and
+        Andersen, {\\O}istein E.  and
+        Briscoe, Ted",
+    booktitle = "Proceedings of the Fourteenth Workshop on Innovative Use of NLP for Building Educational Applications",
+    month = aug,
+    year = "2019",
+    address = "Florence, Italy",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/W19-4406",
+    doi = "10.18653/v1/W19-4406",
+    pages = "52--75",
+    abstract = "This paper reports on the BEA-2019 Shared Task on Grammatical Error Correction (GEC). As with the CoNLL-2014 shared task, participants are required to correct all types of errors in test data. One of the main contributions of the BEA-2019 shared task is the introduction of a new dataset, the Write{\\&}Improve+LOCNESS corpus, which represents a wider range of native and learner English levels and abilities. Another contribution is the introduction of tracks, which control the amount of annotated data available to participants. Systems are evaluated in terms of ERRANT F{\\_}0.5, which allows us to report a much wider range of performance statistics. The competition was hosted on Codalab and remains open for further submissions on the blind test set.",
+}
+"""
+
+_DESCRIPTION = """\
+Write & Improve (Yannakoudakis et al., 2018) is an online web platform that assists non-native
+English students with their writing. Specifically, students from around the world submit letters,
+stories, articles and essays in response to various prompts, and the W&I system provides instant
+feedback. Since W&I went live in 2014, W&I annotators have manually annotated some of these
+submissions and assigned them a CEFR level.
+"""
+
+_HOMEPAGE = "https://www.cl.cam.ac.uk/research/nl/bea2019st/#data"
+
+_LICENSE = ""
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "https://www.cl.cam.ac.uk/research/nl/bea2019st/data/wi+locness_v2.1.bea19.tar.gz"
+
+
+class WriteImproveLocness(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class WiLocness(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `wi_locness`

you'll have to regenerate the dataset_infos.json file after this change